packages/graph/hash_graph/lib/interval-ops/src/lib.rs

```diff
@@ -0,0 +1,183 @@
+#![no_std]
+// Not required, reason: code quality
+#![feature(lint_reasons)]
+// Used to canonicalize intervals
+#![cfg_attr(feature = "canonicalize", feature(step_trait))]
```

Comment:
What is/was the reasoning behind the `canonical` name? :)

Reply:
Originally, I had an implementation, which is only used for discrete values, such as integers. The main difference was to have unified bounds (left-closed-right-opened). It's probably worth just removing the struct again.